### PR TITLE
ContainerPilot 3.0.0-RC1

### PR DESCRIPTION
**Pull request by tgross**
This PR updates ContainerPilot to the 3.0.0-RC1 release candidate. Our [Jenkins job](https://product-ci.joyent.us/job/autopilotpattern-consul-containerpilot-v3) is running the tests again and I'll paste those here once they pass.

cc @misterbisson @cheapRoc @geek

### CODE REVIEW

**Comment by tgross** [comment]
Output from https://product-ci.joyent.us/job/autopilotpattern-consul-containerpilot-v3/8/console

```
----------------------------------------------------------------------
ConsulStackFiveNodeTest.test_graceful_leave
----------------------------------------------------------------------
elapsed  | task
17.49011 | docker-compose -f docker-compose.yml -p consul stop
13.45526 | docker-compose -f docker-compose.yml -p consul rm -f
27.91134 | docker-compose -f docker-compose.yml -p consul up -d
1.409824 | docker-compose -f docker-compose.yml -p consul ps
38.14145 | docker-compose -f docker-compose.yml -p consul scale consul=5
2.338416 | docker-compose -f docker-compose.yml -p consul ps
2.338950 | settle: 5
0.778185 | docker inspect consul_consul_1
28.64325 | converge: 5
15.85217 | docker stop consul_consul_3
15.75583 | docker stop consul_consul_4
20.73350 | docker stop consul_consul_5
2.685496 | docker-compose -f docker-compose.yml -p consul ps
2.686041 | settle: 2
.----------------------------------------------------------------------
ConsulStackFiveNodeTest.test_quorum_consistency
----------------------------------------------------------------------
elapsed  | task
17.04216 | docker-compose -f docker-compose.yml -p consul stop
14.56441 | docker-compose -f docker-compose.yml -p consul rm -f
28.10858 | docker-compose -f docker-compose.yml -p consul up -d
1.371529 | docker-compose -f docker-compose.yml -p consul ps
33.25523 | docker-compose -f docker-compose.yml -p consul scale consul=5
2.429924 | docker-compose -f docker-compose.yml -p consul ps
2.430318 | settle: 5
0.669121 | docker inspect consul_consul_1
2.040930 | converge: 5
2.497306 | docker exec consul_consul_3 ifconfig eth0 down
2.700992 | docker exec consul_consul_4 ifconfig eth0 down
2.601500 | docker exec consul_consul_5 ifconfig eth0 down
2.652927 | docker exec consul_consul_5 curl --fail -XPUT -d someval localhost:8500/kv/somekey
2.507064 | docker exec consul_consul_3 ifconfig eth0 up
2.680277 | docker exec consul_consul_4 ifconfig eth0 up
2.415740 | docker exec consul_consul_5 ifconfig eth0 up
1.293207 | converge: 5
./usr/lib/python3.5/_collections_abc.py:595: ResourceWarning: unclosed <socket.socket fd=4, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('172.17.0.2', 54798), raddr=('165.225.167.157', 8500)>
  return self[key]
----------------------------------------------------------------------
ConsulStackFiveNodeTest.test_rejoin_raft
----------------------------------------------------------------------
elapsed  | task
22.88540 | docker-compose -f docker-compose.yml -p consul stop
14.69262 | docker-compose -f docker-compose.yml -p consul rm -f
27.87393 | docker-compose -f docker-compose.yml -p consul up -d
1.387382 | docker-compose -f docker-compose.yml -p consul ps
33.63550 | docker-compose -f docker-compose.yml -p consul scale consul=5
2.400283 | docker-compose -f docker-compose.yml -p consul ps
2.400731 | settle: 5
0.621151 | docker inspect consul_consul_1
8.886080 | converge: 5
20.75455 | docker restart consul_consul_1
2.300987 | docker-compose -f docker-compose.yml -p consul ps
2.301410 | settle: 5
0.214476 | converge: 5
.----------------------------------------------------------------------
ConsulStackThreeNodeTest.test_rejoin_raft
----------------------------------------------------------------------
elapsed  | task
23.11086 | docker-compose -f docker-compose.yml -p consul stop
14.73134 | docker-compose -f docker-compose.yml -p consul rm -f
27.94680 | docker-compose -f docker-compose.yml -p consul up -d
1.397397 | docker-compose -f docker-compose.yml -p consul ps
28.03193 | docker-compose -f docker-compose.yml -p consul scale consul=3
1.948631 | docker-compose -f docker-compose.yml -p consul ps
1.949080 | settle: 3
0.666693 | docker inspect consul_consul_1
11.24025 | converge: 3
20.83942 | docker restart consul_consul_1
1.890587 | docker-compose -f docker-compose.yml -p consul ps
1.890943 | settle: 3
0.253802 | converge: 3
.
----------------------------------------------------------------------
Ran 4 tests in 579.044s

OK
```

File: etc/containerpilot.json5
Comment (geek) [inline etc/containerpilot.json5]: should we make this configurable, coming from an env variable?

Comment (tgross) [reply etc/containerpilot.json5]: Maybe, but that'd be under separate work.

**Review by geek — APPROVED** [review]

**Comment by tgross** [comment]
Any objections to merging this?

File: etc/containerpilot.json5
Comment (misterbisson) [reply etc/containerpilot.json5]: The problem is that this moves the `-bootstrap-expect 3` from the `docker-compose.yml` (where it is easily configured) into a place where it isn't.

That feature (being able to switch from demo-friendly non-HA to production-friendly HA) is being used in other blueprints.

Comment (tgross) [reply etc/containerpilot.json5]: Fair enough. We can't put the exec in the Compose manifest anymore, but we can do something with the env vars.

Comment (tgross) [reply etc/containerpilot.json5]: That being said -- why use this blueprint at all for a single node? This blueprint is kind of a poor demo for that reason.

Comment (tgross) [reply etc/containerpilot.json5]: When I went to go do the env vars I realized we _do_ have a 3-node demo in the `local-compose.yml`. So really we're just talking about reusing this instead of `consul:latest` for being consumed by other blueprints as a one-node. For demos we _really_ want to use `-dev` mode for Consul rather than `--bootstrap-expect=1` (for time-to-launch purposes), so I'll try to crowbar that into the env var setup here somehow.

Comment (tgross) [reply etc/containerpilot.json5]: I've pushed this commit https://github.com/autopilotpattern/consul/pull/37/commits/4a9acb7232fa18771fd671396a6a91de6e9141d9 which should cover us. What do we think of that?

**Comment by tgross** [comment]
As per https://github.com/autopilotpattern/consul/pull/37#discussion_r119858459 I think I've got the last of the objections covered. Ready to merge if I get a LGTM

**Review by misterbisson — APPROVED** [review]
:pickles:

**Pull request merged**
